ui/components/app/gas-customization/advanced-gas-inputs/advanced-gas-inputs.component.js

```diff
@@ -52,22 +52,32 @@ export default class AdvancedGasInputs extends Component {
     }
   }
 
-  onChangeGasLimit = (e) => {
-    this.setState({ gasLimit: e.target.value });
-    this.changeGasLimit({ target: { value: e.target.value } });
+  onChangeGasLimit = ({ target: { value } }) => {
+    const newGasLimit = Number(value);
+    if (isNaN(newGasLimit) || newGasLimit < 0) {
```

Comment:
It might be worth preventing the user from setting this to non-integer values as well. I think only integers are valid here.

Reply:
sorry, just want to make sure I understand, are you suggesting something additional or are you following up on https://github.com/MetaMask/metamask-extension/pull/12061#discussion_r706216696

Reply:
This would be something additional. I don't mean to block the PR - what you have here is still an improvement - but I think "Ensuring only valid numerical inputs" should include disallowing non-integer numbers. e.g. we can't allow the gas limit to be "0.5" or "3.1415".

Reply:
The same applies to the gas price, except that for the price we can allow a certain number of decimal places. I can't recall how many :thinking: Basically we want the value in WEI to be an integer. But IIRC the input is in GWEI

Reply:
Ah I understand! I'd be happy to update this PR to include that check, may as well do it now :)